### PR TITLE
DB-7267 Support HDP 2.6.4

### PR DESCRIPTION
**Pull request by perryluopl**
Done.

### CODE REVIEW

File: assembly/hdp2.6.4/build/var/lib/ambari-server/resources/common-services/SPLICEMACHINE/2.5.1/service_advisor.py [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+    hdfs_site_desired_values = {
+        "dfs.datanode.handler.count" : "20",
+        "dfs.client.read.shortcircuit.buffer.size" : "131072",
+
+    }
+    return hdfs_site_desired_values
+
+  def getHBaseSiteDesiredValues(self):
+    hbase_site_desired_values = {
+        "hbase.coprocessor.master.classes" : "org.apache.ranger.authorization.hbase.RangerAuthorizationCoprocessor,com.splicemachine.hbase.SpliceMasterObserver",
```
Comment (wb14123) [inline assembly/hdp2.6.4/build/var/lib/ambari-server/resources/common-services/SPLICEMACHINE/2.5.1/service_advisor.py]: DB-7222 removed `org.apache.ranger.authorization.hbase.RangerAuthorizationCoprocessor`, this PR need to remove this, too.

File: assembly/hdp2.6.4/build/var/lib/ambari-server/resources/common-services/SPLICEMACHINE/2.5.1/service_advisor.py [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+        "hbase.regionserver.global.memstore.size" : "0.25",
+        "hfile.block.cache.size" : "0.25",
+        "hbase.regionserver.handler.count" : "200",
+        "hbase.client.scanner.caching" : "1000",
+        "hbase.hstore.blockingStoreFiles" : "20",
+        "hbase.hstore.compactionThreshold" : "5",
+        "hbase.balancer.period" : "60000",
+        "hbase.client.ipc.pool.size" : "10",
+        "hbase.client.max.perregion.tasks" : "100",
+        "hbase.coprocessor.regionserver.classes" : "org.apache.ranger.authorization.hbase.RangerAuthorizationCoprocessor,com.splicemachine.hbase.RegionServerLifecycleObserver",
```
Comment (wb14123) [inline assembly/hdp2.6.4/build/var/lib/ambari-server/resources/common-services/SPLICEMACHINE/2.5.1/service_advisor.py]: DB-7222 removed `org.apache.ranger.authorization.hbase.RangerAuthorizationCoprocessor`

File: assembly/hdp2.6.4/build/var/lib/ambari-server/resources/common-services/SPLICEMACHINE/2.5.1/service_advisor.py [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+  def getServiceComponentLayoutValidations(self, services, hosts):
+      print "getServiceComponentLayoutValidations"
+      return []
+
+
+  def getServiceConfigurationRecommendations(self, configurations, clusterData, services, hosts):
+    #Update HBase Classpath
+    print "getServiceConfigurationRecommendations",services
+    splice_jars = ":".join([jar for jar in glob.glob('/var/lib/splicemachine/*.jar')])
+    spark_jars = ":".join([jar for jar in glob.glob('/usr/hdp/2.6.3.0-235/spark2/jars/*.jar')])
```
Comment (wb14123) [inline assembly/hdp2.6.4/build/var/lib/ambari-server/resources/common-services/SPLICEMACHINE/2.5.1/service_advisor.py]: `2.6.3.0-235` in the path need to be changed to match the version

File: pom.xml [hunk 4]
```diff
@@ -933,7 +933,33 @@
...
+            <activation>
+                <property>
+                    <name>env</name>
+                    <value>hdp2.6.4</value>
+                </property>
+            </activation>
+            <properties>
+                <envClassifier>hdp2.6.4</envClassifier>
+                <envHbase>hbase1.1.2.2.6</envHbase>
+                <hadoop.version>2.7.3.2.6.3.0-235</hadoop.version>
```
Comment (wb14123) [inline pom.xml]: This version may need to match hdp 2.6.4.

**Comment by perryluopl** [comment]
Fixed. Removed org.apache.ranger.authorization.hbase.RangerAuthorizationCoprocessor. Change to 2.7.3.2.6.4.0-91 for hadoop in pom.xml

File: pom.xml [hunk 5]
```diff
@@ -933,7 +933,33 @@
...
+            <activation>
+                <property>
+                    <name>env</name>
+                    <value>hdp2.6.4</value>
+                </property>
+            </activation>
+            <properties>
+                <envClassifier>hdp2.6.4</envClassifier>
+                <envHbase>hbase1.1.2.2.6</envHbase>
+                <hadoop.version>2.7.3.2.6.4.0-91</hadoop.version>
```
Comment (wtang16) [inline pom.xml]: How is 2.7.3.2.6.4.0-91 chosen? Is this the latest stable version?

There are other version available from Hortonworks repo:
http://repo.hortonworks.com/content/repositories/releases/org/apache/hadoop/hadoop-main/

**Pull request merged**